apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx

```diff
@@ -247,8 +264,20 @@ export function ActivityDetails(props: {
     outputToken = COW[chainId]
   }
 
+  const isCustomRecipient = order?.owner !== order?.receiver
+
   return (
     <>
+      {/* Warning banner if custom recipient */}
+      {order && (isCreating || isPending || isPresignaturePending) && isCustomRecipient && isCustomRecipientWarningBannerVisible && order.receiver && <CustomRecipientWarningBanner
```

Comment:
@alfetopito Updated to make sure it only shows the warning banner on specific states (unless dismissed)

<img width="970" alt="Screenshot 2023-09-13 at 14 14 53" src="https://github.com/cowprotocol/cowswap/assets/31534717/ca1e425f-fca8-4a30-8dc1-5426e688444e">